docs/source/daml2ts/index.rst

```diff
@@ -0,0 +1,291 @@
++-------------------+------------------+----------------------------------+---------------------------------------------------------------------+
+| ``Optional<T>``   | ``Optional t``   | ``null | (null extends T ?``     | ``Optional: (t: Serializable<T>) => Serializable<Optional<T>>``     |
+|                   |                  | ``[] | [Exclude<T, null>] : T)`` |                                                                     |
++-------------------+------------------+----------------------------------+---------------------------------------------------------------------+
+| ``TextMap<T>``    | ``TextMap t``    | ``{ [key: string]: T }``         | ``TextMap: (t: Serializable<T>) => Serializable<TextMap<T>>``       |
++-------------------+------------------+----------------------------------+---------------------------------------------------------------------+
+
+DAML to Typescript mappings
+---------------------------
+
+The mappings from DAML to Typescript are best explained by example.
+
+Records
+~~~~~~~
+
+Basic
+*****
+
+In DAML, we might model a person like this.
+
+.. code-block:: daml
+   :linenos:
+
+   data Person =
+     Person
+       name : Text
+       party : Party
+       age : Int
+
+Given the above definition, the generated Typescript code will be as follows.
+
+   .. code-block:: typescript
+      :linenos:
+
+      export type Person = {
+        name: string;
+        party: daml.Party;
+        age: daml.Int;
+      }
+
+      export const Person: daml.Serializable<Person> = ({
+        decoder: () => jtv.object({
+          name: daml.Text.decoder(),
+          party: daml.Party.decoder(),
+          aage: daml.Int.decoder(),
+        }),
+      })
+
+Generic
+*******
+
+In the above case, ``Person`` is a monomorphic type. If we change it slightly so that it is polymorphic in the type of ``age``, it would look like this.
+
+.. code-block:: daml
+   :linenos:
+
+   data Person a =
+     Person
+       name : Text
+       party : Party
+       age : a
+
+In this case, the generated Typescript would read like the following.
+
+.. code-block:: typescript
+   :linenos:
+
+   export type Person<a> = {
+     name: string;
+     party: daml.Party;
+     age: a;
+   }
+
+   export const Person = <a>(a: daml.Serializable<a>): daml.Serializable<Person<a>> => ({
+     decoder: () => jtv.object({
+       name: daml.Text.decoder(),
+       party: daml.Party.decoder(),
+       age: a.decoder(),
+     }),
+   })
+
+Tuples
+~~~~~~
+
+Suppose the following DAML definition.
+
+.. code-block:: daml
+   :linenos:
+
+   data Pair a b = Pair (a, b)
+
+In Typescript, the above definition renders as follows.
+
+.. code-block:: typescript
+   :linenos:
+
+   import * as pkg40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7_DA_Types from './../40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7/DA/Types';
```

Comment:
I think this could use an explanation that `40…f7` is the package id. I would also recommend to import it as `pkg_…` to make it clear that `pkg` is just a prefix (presumably that prefix is needed since the identifier cannot start with a number?

Reply:
This is going to go away. After consultation with Martin we have decided to gloss over some detail of which this will be one.